packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts

```diff
@@ -194,38 +194,26 @@ async function generateInitialScripts(
   }
 
   const scripts: HtmlTagDescriptor[] = [];
-  let injectedBlockingViteClient = false;
 
   for (const result of sources.values()) {
-    // CSS modules do not support result.source (empty) since ssr-transforming them gives the css keys
-    // and client-transforming them gives the script tag for injecting them.
     if (result.id.endsWith('.module.css')) {
-      if (!injectedBlockingViteClient) {
-        // since we use the client-transformed script tag, we need to avoid FOUC by parse-blocking the vite client that the script imports
-        // this way we make sure to run the CSS modules script tag before everything
-        // blocking this way is not ideal but it works. It should be revisited.
-        scripts.push({
-          tag: 'script',
-          attrs: { type: 'module', blocking: 'render', src: '/@vite/client' },
-          injectTo: 'head-prepend',
-        });
-        injectedBlockingViteClient = true;
-      }
+      // CSS modules do not support result.source (empty) since ssr-transforming them gives the css keys
+      // and client-transforming them gives the script tag for injecting them.
       scripts.push({
         tag: 'script',
-        // tried render blocking this script tag by data url imports but it gives `/@vite/client: Invalid relative url or base scheme isn't hierarchical.` which could not find a way to fix.
-        attrs: { type: 'module', 'waku-module-id': result.id },
+        // async: true to wait vite/client to be loaded
+        attrs: { type: 'module', async: true, 'waku-module-id': result.id },
```

Comment:
I guess this would cause FOUC! Since css will potentially load later.

Reply:
yeah, but better than waiting everything.

Reply:
so, if there are a better solution, it's better.

Reply:
On another thought, I'm not sure if it really causes FOUC. Let's reconsider if it happens.

Reply:
Yeah, it causes FOUC. I found a better fix: #659 
btw `async: true` makes the execution earlier.